lib/node_types/esp/src/device-manager.h

```diff
@@ -12,61 +88,264 @@
 #include <toolbox.h>
 #include <device.h>
 
-// scheduler
+/**
+ * DO_LATER SCHEDULER - NON-BLOCKING DELAYED EXECUTION
+ * ====================================================
+ * The do_later functions provide a simple way to schedule callbacks for future execution.
+ * This is essential for IoT applications that need timing without blocking.
+ * 
+ * TWO VARIANTS
+ * ============
+ * 
+ * 1. WITH ID (Replaceable callbacks):
+ *    do_later(delay_ms, id, callback)
+ *    - If a callback with the same id is already scheduled, it's replaced
+ *    - Useful for debouncing or ensuring only one pending action
+ *    - Example: Button press scheduling repeated callbacks - only latest matters
+ * 
+ * 2. WITHOUT ID (One-shot callbacks):
+ *    do_later(delay_ms, callback)
+ *    - Creates a new scheduled callback
+ *    - Multiple callbacks can be scheduled simultaneously
+ *    - Example: Turn on multiple devices at different times
+ * 
+ * COMPARISON TO STANDARD ARDUINO
+ * ==============================
+ * 
+ * Standard Arduino (blocking):
+ *   button_pressed = digitalRead(BUTTON_PIN);
+ *   if (button_pressed) {
+ *     delay(5000);  // BLOCKS EVERYTHING!
+ *     led.off();
+ *   }
+ * 
+ * Standard Arduino (non-blocking but complex):
+ *   unsigned long button_time = 0;
+ *   bool waiting = false;
+ *   
+ *   void loop() {
+ *     if (digitalRead(BUTTON_PIN)) {
+ *       button_time = millis();
+ *       waiting = true;
+ *     }
+ *     if (waiting && millis() - button_time >= 5000) {
+ *       led.off();
+ *       waiting = false;
+ *     }
+ *   }
+ * 
+ * With IoTempower do_later:
+ *   if (button.is("pressed")) {
```

Comment:
you would usually use IoTempower's IN notation here IN(button).is(pressed) expanding button to its internal name

Same for led: IN(led).value().on() (also using the abbreviating method on instead of from Ustring notation)